llama_index/multi_modal_llms/base.py

```diff
@@ -0,0 +1,137 @@
+from abc import abstractmethod
+from enum import Enum
+from typing import Any, AsyncGenerator, Generator, Optional, Sequence
+
+from llama_index.bridge.pydantic import BaseModel, Field
+from llama_index.constants import (
+    DEFAULT_CONTEXT_WINDOW,
+    DEFAULT_NUM_INPUT_FILES,
+    DEFAULT_NUM_OUTPUTS,
+)
+from llama_index.schema import BaseComponent, ImageDocument
```

Comment:
some context about `Document` and `ImageDocument` object here.

So far, we've made a distinction between `Document` and `Node` objects in LlamaIndex: 
* document is the output of a data loader (which represents the original documents, .e.g. correspond to a markdown file etc)
* node is the intermediate unit of data (that's typically result of chunking, metadata enrichment, used for embedding etc).

I have some small hesitation on directly re-using the `ImageDocument` representation here.

Reply:
we should probably come up with some Image loader abstraction though

Reply:
to clarify I understand/agree that:
For current `Document` or `ImageDocument` in LlamaIndex, as @Disiok said, they are mostly used for output of a data loader.

For this MM model, the images are the input to the modal not the output. 
Agree that better not to use `ImageDocument` for both our model input and RAG output to avoid confusion

Reply:
ya exactly, basically what I had in mind was something along the lines of `ImagePrompt` of `MultimodalMessage` that serves as the type for LLM inputs (i.e. to mirror text prompt, chat messages).

This is more decoupled with the types used for document loading, but at the expense of introducing more classes. 

You can make the final call for the PR. but maybe add a note that the interface may change as we iterate.